bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -276,11 +285,33 @@ public CloudBigtableScanConfiguration build() {
         if (scan == null) {
           scan = new Scan();
         }
-        ReadRowsRequest.Builder builder = Adapters.SCAN_ADAPTER.adapt(scan, readHooks);
-        request =  StaticValueProvider.of(readHooks.applyPreSendHook(builder.build()));
+        String materializedProjectId = PLACEHOLDER_PROJECT_ID;
+        String materializedInstanceId = PLACEHOLDER_INSTANCE_ID;
+        String materializedAppProfileId = PLACEHOLDER_APP_PROFILE_ID;
+
+        if(projectId != null && Strings.isNullOrEmpty(projectId.get())){
```

Comment:
Please don't call `.get()` here.  Please assume that projectId & etc are not available.  Please see `CloudBigtableConfiguration.getDisplayData()` for proper checking of value provider availability.

Reply:
Updated code, Now adapting through place holder for all four values i.e. projectId, instanceId, appProfileId & TableId.